File: python/18_flatten_one_level.py
"""Aplanar un nivel — Fácil · Arrays

Retorna una nueva lista eliminando únicamente las sub-listas de primer nivel.
No hagas un aplanamiento recursivo ni uses itertools.chain.

Fuente: https://coding-challenges.dev/problems/python-aplanar-un-nivel
"""


def flatten_one_level(array: list) -> list:
    flattened = []
    for item in array:
        flattened.extend(item if isinstance(item, list) else [item])
    return flattened

File: typescript/18-flatten-one-level.ts
// Aplanar un nivel — Fácil · Arrays
// Aplana solo los sub-arrays de primer nivel; no uses Array.prototype.flat().
// Fuente: https://coding-challenges.dev/problems/aplanar-un-nivel

function flattenOneLevel(array: (number | number[])[]): number[] {
  const flattened: number[] = [];
  for (const item of array) {
    if (Array.isArray(item)) flattened.push(...item);
    else flattened.push(item);
  }
  return flattened;
}

// No modificar: necesario para evaluar el resultado.
export { flattenOneLevel };
